bentoml/spacy.py

```diff
@@ -391,24 +413,57 @@ def _setup(self) -> None:  # type: ignore[override]
             config=self._config,
         )
 
-    # pylint: disable=arguments-differ
-    def _run_batch(  # type: ignore[override]
+    @overload  # pragma: no cover
+    def _run_batch(  # pylint: disable=arguments-differ
+        self,
+        args: t.Iterable[t.Union[str, "Doc"]],  # type: ignore[reportUnknownParameterType]
+        *,
+        as_tuples: Literal[False] = ...,
+        batch_size: t.Optional[int] = ...,
+        disable: t.Iterable[str] = ...,
+        component_cfg: t.Optional[t.Dict[str, t.Dict[str, t.Any]]] = ...,
+        n_process: int = ...,
+    ) -> t.Iterator["Doc"]:  # type: ignore[reportUnknownParameterType]
+        ...
+
+    @overload  # pragma: no cover
+    def _run_batch(  # noqa: F811 # pylint: disable=arguments-differ
         self,
-        inputs: t.Iterable[t.Tuple[str, t.Any]],
-        as_tuples: bool = False,
-        batch_size: t.Optional[int] = None,
-        disable: t.Iterable[str] = SimpleFrozenList(),  # noqa
-        component_cfg: t.Optional[t.Dict[str, t.Dict[str, t.Any]]] = None,
-    ) -> t.Iterator[t.Tuple[t.Union["Doc", "Doc"], t.Any]]:
-        if not isinstance(inputs, list):
-            inputs = list(inputs)
-        return self._model.pipe(
-            inputs,
-            as_tuples=as_tuples,
-            batch_size=batch_size,
-            disable=disable,
-            component_cfg=component_cfg,
-            n_process=self.num_replica,
+        args: t.Iterable[t.Tuple[t.Union[str, "Doc"], t.Any]],  # type: ignore[reportUnknownParameterType] # noqa: LN001
+        *,
+        as_tuples: Literal[True] = ...,
```

Comment:
I think all these should be `partial_args` passed in from `load_runner`,  they probably should not be args of `_run_batch`

Reply:
every parameter in `_run_batch` goes through a data container batch process, so it will never be type `t.Optional[int]`. Supported parameter types are only the ones registered with DataConatiner:  https://github.com/bentoml/BentoML/blob/bentoml-1.0/bentoml/_internal/runner/container.py#L265 - basically numpy, pandas, t.List, and we are adding a few framework-specific ones including tf.Tensor and pytorch tensor